backend/verifai_service.py

```diff
@@ -205,13 +246,14 @@ def search(query: Query):
 
 
 @app.post("/query")
-async def answer_generation(query: Query):
+async def answer_generation(query: Query, Authorize: AuthJWT = Depends()):
     global documents_found
+    Authorize.jwt_required()
```

Comment:
That function ensure that the token is valid when arrives a request HTTP from frontend side.
This is the documentation of the code [https://indominusbyte.github.io/fastapi-jwt-auth/api-doc/](https://indominusbyte.github.io/fastapi-jwt-auth/api-doc/).

The default parameter in `auth_from `is "request", which means that it will verify the token by the headers.
The HEADERS OF HTTP request should be this: "Bearer " + **TOKEN**

Reply:
Yes, I get that, but if user wants to use it in different app, how it will get API token? And if he/she gets token using this method, it will be active only for few hours. We discussed to have tokens in user database that user can get and use permanently. That I think is missing

Reply:
Now I have created a custom Method (on fast api security) to verify the Token. 
The token is stored in a database,  after the request the codechecks if the token exists in the database.
This allow a verification in the App (frontend side) but also a verification if someone else wants to use the API.